### PR TITLE
Improve unmount for snapshot

### PR DESCRIPTION
**Pull request by yanxuean**
ref: https://github.com/containerd/containerd/issues/1785
1. sync Unmount for snapshot
2. best effort to remove temp dir for snapshot

Signed-off-by: yason <yan.xuean@zte.com.cn>

### CODE REVIEW

File: pkg/containerd/opts/container.go [hunk 1]
```diff
@@ -71,13 +71,21 @@ func WithVolumes(volumeMounts map[string]string) containerd.NewContainerOpts {
...
+		// We change RemoveAll to Remove so that we either leak a temp dir
+		// if it fails but not RM snapshot data.
+		// refer to https://github.com/containerd/containerd/pull/1868
+		// https://github.com/containerd/containerd/pull/1785
+		defer os.Remove(root) // nolint: errcheck
+		if err := mount.All(mounts, root); err != nil {
+			return errors.Wrap(err, "failed to mount")
 		}
-		defer unix.Unmount(root, 0) // nolint: errcheck
+		defer func() {
```
Comment (Random-Liu) [inline pkg/containerd/opts/container.go]: I prefer we log the error here in our code.
Or else if `err != nil`, we'll lose the error info.

Comment (yanxuean) [reply pkg/containerd/opts/container.go]: Yes. It is same with happened in containerd. When i modify it in containerd, I don't know how to aggregate these errors. So...
I will find the way to aggregate it again.

Comment (Random-Liu) [reply pkg/containerd/opts/container.go]: I think log it is fine. :)

Comment (yanxuean) [reply pkg/containerd/opts/container.go]: done

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**